src/lib/colorUtils.ts

```diff
@@ -0,0 +1,11 @@
+import * as _ from 'lodash'
+import { SiteVariablesInput, ColorVariants, ColorValues } from '../themes/types'
+
+export const mapColorsToScheme = <T>(
+  siteVars: SiteVariablesInput,
+  mapper: keyof ColorVariants | ((color: ColorVariants) => T),
+): ColorValues<T> =>
+  _.mapValues(
+    { ...siteVars.emphasisColors, ...siteVars.naturalColors },
```

Comment:
@layershifter 
thinking one more time about this, rather than merging always these 2 objects, why don't we just use the
`siteVars.colors` that contains already this merge:

https://github.com/stardust-ui/react/blob/4a7eceae3b3f0571a667c8622eb87d7b0cdf00d5/src/themes/default/colors.ts#L138
```jsx
export const colors: ColorPalette = {
  ...contextualColors,
  ...emphasisColors,
  ...naturalColors,

  black: '#000',
  white: '#fff',
}
```
we'd need `contextualColors` anyways when deciding upon text colors for box components like `Button` or `Label`:
```jsx
Label color='red' ../>
```

```jsx
Label color='white' ../>
```
will apply
```jsx
const bgColor = siteVars.colors['white'][500], // or other variant
const textColorVariant = Color(bgColor).isDark()
  ? 50 /* white text */
  : 900 /* black text (or other variants) */
{
  backgroundColor: bgColor,
  color: siteVars.colors.text[textColorVariant],
}
```

What do you think about:

```suggestion
    siteVars.colors,
```
and the other approach using `Color(bgColor).isDark()`?

Reply:
> siteVars.colors that contains already this merge:

From one side I will agree with, from other side `colors` contains colors from `contextualColors` like `text` and `danger`. We considered previously that `danger` is a state of a component, not a color, i.e:

```jsx
<Message danger /> // valid
<Message color='danger' /> // invalid
```

Reply:
We can use `Color()`, i.e the `color` package for docs needs. But we should omit it as potential dependency, #583.

Reply:
> from other side colors contains colors from contextualColors like text and danger
I'd like to avoid recreating this object every time the method is called. What about creating this `{ ...siteVars.emphasisColors, ...siteVars.naturalColors },` object in the theme object directly, the same way we did [here](https://github.com/stardust-ui/react/blob/4a7eceae3b3f0571a667c8622eb87d7b0cdf00d5/src/themes/default/colors.ts#L138) for colors object?

Reply:
> We can use Color(), i.e the color package for docs needs. But we should omit it as potential dependency, #583.

how else can we use it if we want to omit it as potential dependency? 😕

Reply:
I mean that we can use it as dev dependency, but `color` can not be used as a package dependency, i.e. it can be a part of lib or themes.